server/src/main/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorService.java

```diff
@@ -79,22 +83,26 @@ public HealthIndicatorResult calculate() {
             }
         }
 
-        return createIndicator(status.getStatus(), status.getSummary(), status.getDetails());
+        return createIndicator(status.getStatus(), status.getSummary(), status.getDetails(), status.getImpact());
     }
 
     private static class ShardAllocationCounts {
-        private boolean available = true;
+        private boolean available = true; // This will be true even if no replicas are expected, as long as none are unavailable
```

Comment:
I think this comment is not valid anymore?

Reply:
Hmm, yeah that last part is. I'll fix it.